src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherFactory.java

```diff
@@ -34,16 +36,30 @@ public GitHubChecksPublisherFactory() {
 
     @Override
     protected Optional<ChecksPublisher> createPublisher(final Run<?, ?> run) {
-        Optional<GitHubSCMSource> source = scmFacade.findGitHubSCMSource(run);
+        return createPublisher(run.getParent());
+    }
+
+    @Override
+    protected Optional<ChecksPublisher> createPublisher(final Queue.Item item) {
+        if (!(item.task instanceof Job)) {
+            return Optional.empty();
+        }
+
+        return createPublisher((Job<?, ?>)item.task);
+    }
+
+    protected Optional<ChecksPublisher> createPublisher(final Job<?, ?> job) {
```

Comment:
Should we also just have this `createPublisher` which takes a job as parameter instead of providing the other two (use run and queue item) in the checks API plugin?

Also, I have a more general question about such incompatible change for API development: Currently, we just have two consumers, the price is acceptable if changed, but what if consumers become more and more? Should we avoid such incompatible changes at all in that situation? @timja

Reply:
once people start using this plugin we can't make breaking changes really.

well there's a couple of options:

1. release all the plugins at the same time and have users update all of the plugins
2. deprecate the old API, move plugins to the new one, at some point in the future remove it

Reply:
if we only need the one we should probably just have that one, do we not need the run reference though?

Reply:
No run is needed since what we resolved from `run` before is actually resolved from `run.getParent` which is the job.

But we may need a run listener alone with the job later if we want to log to the console.